### ISSUE TITLE
How to Apply parameters to a new yolov8n model

### ISSUE THREAD

**Issue by Darkcoder02** [ultralytics/ultralytics#18530]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello everyone, I am trying to set up a federated learning architecture using **flwr framework**, and to do so I am using the yolov8n model to train in a federated manner. However, I am stuck applying the model weights received from the global server as numpy array to a yolov8n.pt model. I have gone through the documentation and could not find any useful modules to do so.

After I train the model on custom dataset, the weights of the trained model are changed in size and when I try to apply these weights to a yolov8.pt file, I get the size mismatch error. 

PFA: Code to reproduce the error:

```
from ultralytics import YOLO
from collections import OrderedDict
import torch
# Load a model
def load_model(model_name):
    return YOLO(model_name)
    
# train the loaded model
def train_model(model, epochs):
    result = model.train(data='african-wildlife.yaml', epochs=epochs, imgsz=640)
    return result.save_dir

# extract the weights form loaded model
def get_weights(model):
    weight_list = [val.cpu().numpy() for _, val in model.state_dict().items()]
    return weight_list

# apply the weight to the loaded model
def apply_weights(model, weights):
    params_dict = zip(model.state_dict().keys(), weights)
    state_dict = OrderedDict({k: torch.tensor(v) for k, v in params_dict})
    model.load_state_dict(state_dict)    

model = load_model('yolov8n.pt')
path = train_model(model, epochs=1)
weights = get_weights(load_model(path / 'weights' / 'best.pt'))
apply_weights(load_model('yolov8n.pt'), weights)v
```

### Additional

```
<img width="1440" alt="error_ss" src="https://github.com/user-attachments/assets/cd0711ac-7f92-45a3-8467-60c9feb649df" />
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Darkcoder02, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) for helpful resources, where you can find various [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that might provide insights into your issue.

It looks like you're trying to apply custom weights to a YOLOv8 model and encountering a size mismatch problem. If this is a 🐛 Bug Report or unexpected behavior, we kindly ask that you provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) (MRE). This assists us in debugging and understanding the root cause of the problem more effectively. Based on what you've shared, it would be great if you could ensure the code snippet provided is fully self-contained and reproduces the issue when executed in isolation. Additionally, please include a more detailed description of the error message, as well as any relevant logs or screenshots (you've attempted to attach a screenshot, but it appears incomplete or inaccessible).

If this is a custom training ❓ Question, please provide as much context as possible, including details about your federated learning setup, dataset, and any additional modifications you are making to the model. You might also find our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) useful.

## Upgrade

Make sure you’re using the latest version of the `ultralytics` package with all proper dependencies installed. You can update your environment as follows:

```bash
pip install -U ultralytics
```

YOLOv8 is designed to evolve with every release, so upgrading could potentially resolve your issue.

## Environments

You can test and run YOLOv8 in verified, up-to-date environments, including:

- **Notebooks** with free GPU:  
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)  
  <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community Support

💡 Need real-time help? Join the Ultralytics community where it suits you best:  
- [Discord](https://discord.com/invite/ultralytics) for interactive discussions 🎧  
- [Discourse](https://community.ultralytics.com) for longer topics 📘  
- [Subreddit](https://reddit.com/r/Ultralytics) for discussions across the community 👥  

## Status

If you're wondering about the repository's status, you can verify that all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing (indicated by the green badge below). These tests validate YOLOv8's functionality across various modes and platforms daily.

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  

This is an automated response to help you get started quickly. An Ultralytics engineer will review your issue and provide additional support as soon as possible. Stay tuned 🔍!

**Comment by Y-T-G** [comment]
`state_dict()` weights are not used when training because a new model is created.

https://github.com/ultralytics/ultralytics/issues/16520#issuecomment-2380734866

**Comment by Darkcoder02** [comment]
> `state_dict()` weights are not used when training because a new model is created.
> 
> https://github.com/ultralytics/ultralytics/issues/16520#issuecomment-2380734866

So can I use the extracted weights from my best.pt model, which get after training, for loading it into a default yolov8.pt model?

**Comment by Y-T-G** [comment]
Read the linked comment
